### PR TITLE
ITF: reworked checker queues

### PR DESCRIPTION
**Pull request by MBoldyrev**
Signed-off-by: Mikhail Boldyrev <miboldyrev@gmail.com>

### Description of the Change

Refactoring. A simple class that wraps the objects that follow a transaction submission in another threads and are waited for and checked later.

### Benefits

Another abstraction makes things clearer. Also removed tbb from ITF, where lock-free performance has lower priority than simplicity and reliability.

### CODE REVIEW

File: test/framework/integration_framework/integration_test_framework.cpp [hunk 1]
```diff
@@ -698,13 +742,16 @@ namespace integration_framework {
...
-                   response,
-                   tx_response_waiting,
-                   "missed status");
+    boost::optional<TxResponseType> opt_response;
+    const auto it = responses_queues_.find(tx_hash.hex());
+    if (it != responses_queues_.end()) {
+      opt_response = it->second->try_pop();
+    }
+    if (not opt_response) {
+      throw std::runtime_error( "missed status");
```
Comment (lebdron) [inline test/framework/integration_framework/integration_test_framework.cpp]:
```suggestion
      throw std::runtime_error("missed status");
```
Looks like file should be clang-formatted.

File: test/framework/integration_framework/integration_test_framework.cpp [hunk 2]
```diff
@@ -95,6 +95,36 @@ namespace {
 
 namespace integration_framework {
 
+  template <typename T>
+  class IntegrationTestFramework::CheckerQueue {
```
Comment (lebdron) [inline test/framework/integration_framework/integration_test_framework.cpp]: Class description should be added.

File: test/framework/integration_framework/integration_test_framework.cpp [hunk 3]
```diff
@@ -664,12 +707,11 @@ namespace integration_framework {
...
     // fetch first proposal from proposal queue
-    VerifiedProposalType verified_proposal_and_errors;
-    fetchFromQueue(verified_proposal_queue_,
-                   verified_proposal_and_errors,
-                   proposal_waiting,
-                   "missed verified proposal");
-    validation(verified_proposal_and_errors->verified_proposal);
+    auto opt_verified_proposal_and_errors = verified_proposal_queue_->try_pop();
+    if (not opt_verified_proposal_and_errors) {
+      throw std::runtime_error( "missed verified proposal");
```
Comment (lebdron) [inline test/framework/integration_framework/integration_test_framework.cpp]:
```suggestion
      throw std::runtime_error("missed verified proposal");
```

File: test/framework/integration_framework/integration_test_framework.cpp (hunk 2 above)
Comment (MBoldyrev) [reply test/framework/integration_framework/integration_test_framework.cpp]: added it to the header file

File: test/framework/integration_framework/integration_test_framework.cpp (hunk 3 above)
Comment (MBoldyrev) [reply test/framework/integration_framework/integration_test_framework.cpp]: thanks, I forgot to run the formatter

**Pull request merged**